src/handle_append_entries.cxx

```diff
@@ -398,7 +446,11 @@ ptr<req_msg> raft_server::create_append_entries_req(ptr<peer>& pp) {
             p.set_next_log_idx(cur_nxt_idx);
         }
 
-        last_log_idx = p.get_next_log_idx() - 1;
+        if (last_streamed_log_idx > 0) {
+            last_log_idx = last_streamed_log_idx;
+        } else {
+            last_log_idx = p.get_next_log_idx() - 1;
+        }
```

Comment:
I use the number here. Because I want to make sure that the **last_streamed_log_idx** won't change during I process the request. When we enter into this function with stream = true, the response may set it to 0 in the middle of processing. Sometimes, the last_log_idx may be 0.

Reply:
Then it will be better to give a generic name `custom_last_log_idx`:
```c++
ptr<req_msg> raft_server::create_append_entries_req(ptr<peer>& pp,
                                                    ulong custom_last_log_idx)
```
and move `p.set_last_streamed_log_idx()` out of `create_append_entries_req`, as I put in the above comment. `create_append_entries_req` is not a proper place to update it (as it is unaware of streaming).